src/main/java/org/openstreetmap/atlas/geography/atlas/items/complex/aoi/ComplexAOI.java

```diff
@@ -0,0 +1,185 @@
+package org.openstreetmap.atlas.geography.atlas.items.complex.aoi;
+
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import org.openstreetmap.atlas.exception.CoreException;
+import org.openstreetmap.atlas.geography.MultiPolygon;
+import org.openstreetmap.atlas.geography.atlas.items.Area;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasEntity;
+import org.openstreetmap.atlas.geography.atlas.items.Relation;
+import org.openstreetmap.atlas.geography.atlas.items.complex.ComplexEntity;
+import org.openstreetmap.atlas.geography.atlas.items.complex.RelationOrAreaToMultiPolygonConverter;
+import org.openstreetmap.atlas.tags.filters.TaggableFilter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+
+/**
+ * Complex Entity for AOI Relations and AOI Areas. AOI relations are those that are
+ * {@link MultiPolygon} and have AOI tags in it and AOI Areas are those {@link Area}s that have AOI
+ * tags in it. AOI tags are checked against the {@link TaggableFilter} passed as an argument or to
+ * the default {@link TaggableFilter} of AOI tags if the tags are not explicitly specified.
+ *
+ * @author sayas01
+ */
+public final class ComplexAOI extends ComplexEntity
```

Comment:
Nit: For those not familiar with the abbreviation, is it better to spell this out at `ComplexAreaOfInterest`?